src/ARC1-Mintable.lua

```diff
@@ -92,14 +90,21 @@ function mint(account, amount)
   _typecheck(amount, 'ubig')
 
   assert(isMinter(system.getSender()), "only minter can mint")
-  assert(not _cap:get() or (_totalSupply:get()+amount) <= _cap:get(), 'totalSupply is over the cap')
+  assert(not _max_supply:get() or (_totalSupply:get()+amount) <= _max_supply:get(), 'totalSupply is over MaxSupply')
 
   _mint(account, amount)
 
   -- contract.event("mint", system.getSender(), account, amount)
   contract.event("transfer", address0, account, amount)
 end
 
+-- return Max Supply
+-- @type    query
+-- @return  amount   (ubig) amount of tokens to mint
+
+function maxSupply()
+  return _max_supply:get()
```

Comment:
it could be this to always return a bignumber:

```lua
  return _max_supply:get() or bignum.number(0)
```

Reply:
...